config/src/keys.rs

```diff
@@ -70,161 +65,55 @@ where
     }
 }
 
-// NetworkKeyPairs is used to store a node's Network specific keypairs.
-// It is filled via a config file at the moment.
-#[derive(Debug, Deserialize, PartialEq, Serialize)]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(Clone))]
-pub struct NetworkKeyPairs {
-    network_signing_private_key: PrivateKeyContainer<Ed25519PrivateKey>,
-    #[serde(serialize_with = "serialize_key")]
-    #[serde(deserialize_with = "deserialize_key")]
-    network_signing_public_key: Ed25519PublicKey,
-
-    #[serde(serialize_with = "serialize_key")]
-    #[serde(deserialize_with = "deserialize_key")]
-    network_identity_private_key: X25519StaticPrivateKey,
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+pub struct KeyPair<T>
+where
+    T: PrivateKey + Serialize + ValidKeyStringExt,
+    T::PublicKeyMaterial: DeserializeOwned + 'static + Serialize + ValidKeyStringExt,
```

Comment:
this is a bit scary tbh :D

Reply:
Ok i see what's happening here and its pretty terrible. This is a consequence of our very overly complicated crypto library. I'm sure we can make this better once we make some improvements to the crypto crate

Reply:
Yeah, there's a few different directions to think about leveraging an Arc::Pin would enabling cloning of configs and keep the keys safely in one space of memory... On the flip side the current API makes that type of change very unfun.